Sources/swift-parsing-benchmark/CSV/CSVBenchmarks.swift

```diff
@@ -2,58 +2,33 @@ import Benchmark
 import Foundation
 import Parsing
 
-//name                 time           std        iterations
-//---------------------------------------------------------
-//CSV.Parser           1245858.500 ns ±  10.38 %       1090
-//CSV.Mutating methods 1309019.000 ns ±   8.27 %       1060
-
-let csvSuite = BenchmarkSuite(name: "CSV") { suite in
-  let rowCount = 1_000
-  let columnCount = 5
-  var output: [[String]] = []
-
-  suite.benchmark(
-    name: "Parser",
-    run: {
-      output = csv.parse(csvInput)!
-    },
-    tearDown: {
-      precondition(output.count == rowCount)
-      precondition(output.allSatisfy { $0.count == columnCount })
-    }
-  )
-
-  suite.benchmark(
-    name: "Ad hoc mutating methods",
-    run: {
-      var input = csvInput[...].utf8
-      output = input.parseCsv()
-    },
-    tearDown: {
-      precondition(output.count == rowCount)
-      precondition(output.allSatisfy { $0.count == columnCount })
-    }
-  )
-}
+//name                        time           std        iterations
+//----------------------------------------------------------------
+//CSV.Parser                  1884886.500 ns ±  13.14 %        694
+//CSV.Ad hoc mutating methods 1229682.000 ns ±  12.79 %       1069
 
 // MARK: - Parser
 
-private typealias Input = Slice<UnsafeBufferPointer<UTF8.CodeUnit>>
+private struct CSV: Parser {
+  func parse(_ input: inout Substring.UTF8View) -> [[String]]? {
```

Comment:
A stylistic change that also slightly improves performance... I think with performance these things may always need to be tested on a case-by-case basis, as this kind of stylistic change doesn't always seem to improve performance (I tried with the date time parser and it slowed things down slightly). Still, it's nice to house the logic in a namespace...